presto-orc/src/test/java/com/facebook/presto/orc/TestReadBloomFilter.java

```diff
@@ -0,0 +1,133 @@
+import static com.facebook.presto.orc.TupleDomainOrcPredicate.ColumnReference;
+import static com.facebook.presto.orc.metadata.CompressionKind.SNAPPY;
+import static com.facebook.presto.spi.predicate.TupleDomain.fromFixedValues;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterables.cycle;
+import static com.google.common.collect.Iterables.limit;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class TestReadBloomFilter
+{
+    private static final TimeZoneKey TIME_ZONE = TimeZoneKey.getTimeZoneKey(HIVE_STORAGE_TIME_ZONE.getID());
+
+    @Test
+    public void test()
+            throws Exception
+    {
+        testType(TINYINT, ImmutableList.of(1L, 50L, 100L), 50L, 77L);
+        testType(SMALLINT, ImmutableList.of(1L, 5000L, 10_000L), 5000L, 7777L);
+        testType(INTEGER, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+        testType(BIGINT, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+
+        // Bloom Filters are not supported for DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR
+
+        testType(DOUBLE, ImmutableList.of(1.11, 500_000.55, 1_000_000.99), 500_000.55, 777_777.77);
+
+        testType(VARCHAR, ImmutableList.of("a", "o", "z"), utf8Slice("o"), utf8Slice("w"));
+        testType(VARBINARY,
+                ImmutableList.of(new SqlVarbinary("a".getBytes(UTF_8)), new SqlVarbinary("o".getBytes(UTF_8)), new SqlVarbinary("z".getBytes(UTF_8))),
+                utf8Slice("o"),
+                utf8Slice("w"));
+    }
+
+    private static <T> void testType(Type type, List<T> uniqueValues, T inBloomFilter, T notInBloomFilter)
+            throws Exception
+    {
+        ArrayList<T> writeValues = newArrayList(limit(cycle(uniqueValues), 30_000));
+
+        try (TempFile tempFile = new TempFile()) {
+            writeOrcColumnHive(tempFile.getFile(), ORC_12, SNAPPY, type, writeValues);
+
+            // without predicate a normal block will be created
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, OrcPredicate.TRUE, type, MAX_BATCH_SIZE)) {
+                assertEquals(recordReader.nextBatch(), 1024);
+            }
+
+            // predicate for specific value within the min/max range without bloom filter being enabled
```

Comment:
these comments are a bit too similar which makes it different to get the difference between the test cases; consider breaking the code into three sections - // no predicate, // predicate for non-matching value, // predicate for matching value - and testing each case with and without the filter like so

```
        try (TempFile tempFile = new TempFile()) {
            writeOrcColumnHive(tempFile.getFile(), ORC_12, SNAPPY, type, writeValues);

            // no predicate
            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.empty(), true)) {
                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
            }

            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.empty(), false)) {
                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
            }

            // predicate for non-matching value
            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(notInBloomFilter), true)) {
                assertEquals(recordReader.nextBatch(), -1);
            }

            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(notInBloomFilter), false)) {
                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
            }

            // predicate for matching value
            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(inBloomFilter), true)) {
                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
            }

            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(inBloomFilter), false)) {
                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
            }
        }


    private static <T> OrcBatchRecordReader createCustomOrcRecordReader(TempFile tempFile, Type type, Optional<T> filterValue, boolean bloomFilterEnabled)
            throws IOException
    {
        OrcPredicate predicate = filterValue.map(value -> makeOrcPredicate(type, value, bloomFilterEnabled)).map(OrcPredicate.class::cast).orElse(TRUE);

        OrcDataSource orcDataSource = new FileOrcDataSource(tempFile.getFile(), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true);
        OrcReader orcReader = new OrcReader(orcDataSource, OrcEncoding.ORC, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));

        assertEquals(orcReader.getColumnNames(), ImmutableList.of("test"));
        assertEquals(orcReader.getFooter().getRowsInRowGroup(), 10_000);

        return orcReader.createBatchRecordReader(ImmutableMap.of(0, type), predicate, HIVE_STORAGE_TIME_ZONE, newSimpleAggregatedMemoryContext(), MAX_BATCH_SIZE);
    }

    private static <T> TupleDomainOrcPredicate<String> makeOrcPredicate(Type type, T value, boolean bloomFilterEnabled)
    {
        return new TupleDomainOrcPredicate<>(
                fromFixedValues(ImmutableMap.of("test", NullableValue.of(type, value))),
                ImmutableList.of(new ColumnReference<>("test", 0, type)),
                bloomFilterEnabled,
                Optional.empty());
    }
```

Reply:
Thanks 
Yes, now the code looks clean and comment is quite simple to understand.